include/fastdds/rtps/common/EntityId_t.hpp

```diff
@@ -154,6 +154,24 @@ struct RTPS_DllAPI EntityId_t
 
 #endif // if !FASTDDS_IS_BIG_ENDIAN_TARGET
 
+    /*!
+     * @brief conversion to uint32_t
+     * @return uint32_t representation
+     */
+    uint32_t to_uint32() const
+    {
+        uint32_t res = *reinterpret_cast<const uint32_t*>(value);
+
+#if !FASTDDS_IS_BIG_ENDIAN_TARGET
+        res = ( res >> 24 ) |
+                (0x0000ff00 & ( res >> 8)) |
+                (0x00ff0000 & ( res << 8)) |
+                ( res << 24 );
+#endif // if !FASTDDS_IS_BIG_ENDIAN_TARGET
+
+        return res;
```

Comment:
We always want the MSB first representation, so we explicitly use it. Most compilers will reduce this to one or two assembly instructions.

```suggestion
        return (static_cast<uint32_t>(value[0]) << 24) |
                (static_cast<uint32_t>(value[1]) << 16) |
                (static_cast<uint32_t>(value[2]) << 8) |
                (static_cast<uint32_t>(value[3]));
```

Reply:
I've already verified with https://godbolt.org/ that is reduced to `bswap eax` in gcc, msvc and clang. Though your approach is less cryptic and I like it.

Reply:
Unfortunately, [msvc](https://godbolt.org/z/47EfE6nb3) doesn't know how to optimize the new approach (gcc and clang do 👍)

Reply:
Ok. Let's leave it the original way then